File: Solutions/1727. Largest Submatrix With Rearrangements/1727. Largest Submatrix With Rearrangements.py
class Solution:
    def largestSubmatrix(self, matrix: List[List[int]]) -> int:
        res = 0
        for row in range(len(matrix)):
            for col in range(len(matrix[0])):
                if matrix[row][col] != 0 and row > 0:
                    matrix[row][col] += matrix[row - 1][col]

            curr=sorted(matrix[row], reverse=True)
            for j in range(len(matrix[0])):
                res = max(res, curr[j] * (j + 1))
            print(curr)
        return res

File: Solutions/1727. Largest Submatrix With Rearrangements/1727. Largest Submatrix With Rearrangements.cpp
class Solution {
public:
    int largestSubmatrix(vector<vector<int>>& matrix) {
        int res=0;
        for(int i=1;i<matrix.size();++i)
            for(int j=0;j<matrix[0].size();++j)
                if(matrix[i][j])
                    matrix[i][j]+=matrix[i-1][j];
        for(int row=0;row<matrix.size();++row){
            sort(begin(matrix[row]),end(matrix[row]),greater());
            for(int col=0;col<matrix[0].size();++col)
                res=max(res,matrix[row][col]*(col+1));
        }
        return res;
    }
};


// update , or copy into new vector
class Solution {
public:
    int largestSubmatrix(vector<vector<int>>& matrix) {
        int res=0;
        for(int i=0;i<matrix.size();++i){
            for(int j=0;j<matrix[0].size();++j){
                if(i>0&&matrix[i][j]!=0)
                    matrix[i][j]=matrix[i-1][j]+1;
            }
            vector<int> foo;
            foo=matrix[i];
            sort(begin(foo),end(foo),greater<int>());
            for(int col=0;col<matrix[0].size();++col)
                res=max(res,(col+1)*foo[col]);
        }
        return res;
    }
};